pkg/snmp/traps/config.go

```diff
@@ -18,13 +21,27 @@ func IsEnabled() bool {
 	return config.Datadog.GetBool("snmp_traps_enabled")
 }
 
+// UserV3 contains the definition of one SNMPv3 user with its username and its auth
+// parameters.
+// TODO: Add support for EngineID differentiation
+type UserV3 struct {
+	Username     string `mapstructure:"user" yaml:"user"`
+	AuthKey      string `mapstructure:"authKey" yaml:"authKey"`
+	AuthProtocol string `mapstructure:"authProtocol" yaml:"authProtocol"`
+	PrivKey      string `mapstructure:"privKey" yaml:"privKey"`
+	PrivProtocol string `mapstructure:"privProtocol" yaml:"privProtocol"`
+}
+
 // Config contains configuration for SNMP trap listeners.
 // YAML field tags provided for test marshalling purposes.
+// TODO: Add namespace
 type Config struct {
-	Port             uint16   `mapstructure:"port" yaml:"port"`
-	CommunityStrings []string `mapstructure:"community_strings" yaml:"community_strings"`
-	BindHost         string   `mapstructure:"bind_host" yaml:"bind_host"`
-	StopTimeout      int      `mapstructure:"stop_timeout" yaml:"stop_timeout"`
+	Port                  uint16   `mapstructure:"port" yaml:"port"`
+	Users                 []UserV3 `mapstructure:"users" yaml:"users"`
+	CommunityStrings      []string `mapstructure:"community_strings" yaml:"community_strings"`
+	BindHost              string   `mapstructure:"bind_host" yaml:"bind_host"`
+	StopTimeout           int      `mapstructure:"stop_timeout" yaml:"stop_timeout"`
+	AuthoritativeEngineID [28]byte `mapstructure:"-" yaml:"-"`
```

Comment:
What's the purpose of `mapstructure:"-" yaml:"-"` for AuthoritativeEngineID ?

AuthoritativeEngineID is always 28 bytes ?

Reply:
This is to prevent storing/reading this from the `datadog.yml` file as this value is generated after reading the config.

It's always 28 bytes in our implementation indeed but can be anywhere between 4 and 32

Reply:
Just having something like below won't work ?
```
type Config struct {
	authoritativeEngineID string
}
```